contracts/cw3-fixed-multisig/src/state.rs

```diff
@@ -57,17 +57,23 @@ impl Proposal {
         self.status = self.current_status(block);
     }
 
-    // returns true iff this proposal is sure to pass (even before expiration if no future
-    // sequence of possible votes can cause it to fail)
-    pub fn is_passed(&self, block: &BlockInfo) -> bool {
+    /// Helper function to check if a certain vote count has reached threshold.
+    /// Only called from is_rejected and is_passed for no and yes votes
+    /// Handles the different threshold types accordingly.
+    /// This function returns true if and only if vote_count is greater than the threshold which
+    /// is calculated.
+    /// In the case where we use yes votes, this function will return true if the proposal will pass.
+    /// In the case where we use no votes, this function will return true if the
+    /// proposal will be rejected regardless of other votes.
+    fn does_vote_count_reach_threshold(&self, vote_count: u64, block: &BlockInfo) -> bool {
         match self.threshold {
             Threshold::AbsoluteCount {
                 weight: weight_needed,
-            } => self.votes.yes >= weight_needed,
+            } => vote_count >= weight_needed,
```

Comment:
This check doesn't work properly of no votes. I would implement it sepately rather than combining.

Assume a threshold of 67%.

If we have 40% yes and 40% no, it would return false for both.
However, this could be safely marked as rejected.

Your logic works fine when the threshold is 50% (same for both).

I would propose not renaming the function and using it for both cases (as done here), but rather copying is_passed to is_rejected and make the various logic changes there needed (often 1-threshold or such). This is different for each branch and hard to generalise. Better to have two separate functions than combine them